scenarioo-client/app/shared/navigation/breadcrumbs/sketcherLink.service.ts

```diff
@@ -15,45 +15,65 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+declare var angular: angular.IAngularStatic;
+
 /**
  * Contains the information about whether to show the sketcher link in the navigation. The link can either be
  * displayed as "Create sketch" or "Edit sketch".
  */
-angular.module('scenarioo.services').factory('SketcherLinkService', SketcherLinkService);
+angular.module('scenarioo.services').factory('SketcherLinkService', SketcherLinkServiceWrapper);
 
-function SketcherLinkService() {
+function SketcherLinkServiceWrapper() {
 
-    var service = this;
+    const service = this;
     service.showCreateOrEditSketchLink = false;
-    service.createOrEditSketchLinkTitle;
-    service.linkClickedAction;
 
     return {
 
-        showCreateOrEditSketchLinkInBreadcrumbs: function(createOrEditSketchLinkTitle, linkClickedAction) {
+        showCreateOrEditSketchLinkInBreadcrumbs(createOrEditSketchLinkTitle, linkClickedAction) {
             service.showCreateOrEditSketchLink = true;
             service.createOrEditSketchLinkTitle = createOrEditSketchLinkTitle;
             service.linkClickedAction = linkClickedAction;
         },
 
-        hideCreateOrEditSketchLinkInBreadcrumbs: function() {
+        hideCreateOrEditSketchLinkInBreadcrumbs() {
             service.showCreateOrEditSketchLink = false;
             service.createOrEditSketchLinkTitle = undefined;
             service.linkClickedAction = undefined;
         },
 
-        isShowCreateOrEditSketchLink: function() {
+        isShowCreateOrEditSketchLink() {
             return service.showCreateOrEditSketchLink;
         },
 
-        getCreateOrEditSketchLinkTitle: function() {
+        getCreateOrEditSketchLinkTitle() {
             return service.createOrEditSketchLinkTitle;
         },
 
-        executeLinkClickedAction: function() {
+        executeLinkClickedAction() {
             service.linkClickedAction();
-        }
-
+        },
     };
+}
+
+export class SketcherLinkService {
+    showCreateOrEditSketchLinkInBreadcrumbs() {
+        return undefined;
+    }
+
+    hideCreateOrEditSketchLinkInBreadcrumbs() {
+        return undefined;
+    }
+
+    isShowCreateOrEditSketchLink() {
+        return undefined;
+    }
+
+    getCreateOrEditSketchLinkTitle() {
+
```

Comment:
couldnt we make the service a real new angular service class and then downgrade it - instead of the other way round ??? I do not like the fact that we start to produce this half migrated services with this ugly classes that do not do anything. I think at least for this simple service here it would have been possible.